examples/pennylane_run_qaoa_maxcut.py

```diff
@@ -0,0 +1,287 @@
+"""
+.. _qaoa_maxcut:
+
+QAOA for MaxCut
+===============
+
+In this tutorial, we demonstrate how to sample joint qubit measurements from a
+variational quantum circuit to solve a combinatorial optimization problem. Namely,
+we implement the quantum approximate optimization algorithm (QAOA) for the MaxCut
+problem as proposed by `Farhi, Goldstone, and Gutmann (2014) <https://arxiv.org/abs/1411.4028>`__
+"""
+
+##############################################################################
+# Background
+# ----------
+#
+# The MaxCut problem
+# ~~~~~~~~~~~~~~~~~~
+# The aim of MaxCut is to maximize the number of edges (yellow lines) in a graph that are "cut" by
+# a given partition of the vertices (blue circles) into two sets as shown in the figure below.
+#
+# .. figure:: ../../examples/figures/qaoa_maxcut_partition.png
+#    :align: center
+#    :scale: 65%
+#    :alt: qaoa_operators
+#
+# |
+# Let's say there are :math:`m` edges. We seek the partition :math:`z` of the vertices into two sets
+# 0 and 1 which maximizes
+#
+# .. math::
+#   C(z) = \sum_{\alpha=1}^{m}C_\alpha(z)
+#
+# where :math:`C_\alpha(z)=1` if :math:`z` places one vertex from the
```

Comment:
At this stage I didn't want to specify the representation of z because I thought it would clutter the explanation of C. Instead, I introduce the bitstring representation in the following paragraph, which has since been updated. (I can consider reordering the paragraphs so that I explain that z can be viewed as a bitstring before I introduce the objective function.)

Reply:
Okay, I've incorporated a short discussion of problem complexity